redux-offline/redux-offline
# [WIP] Add initial jest setup and initial tests

* Add /src/__tests__ directory

* Add Jest and other testing related dev dependencies

* Add jest-setup.js file at project root

* Add tests for middleware, createOfflineStore, config, and batch

I've only got tests in for config, createOfflineStore, middleware, and batch so far, but before I went too much further I wanted to get some feedback and see if heading in the right direction so far.  A few items that I currently have questions about are:

- I am currently putting the `__tests__` directory within the `src` directory. Does that work, or would you rather it be at the root of the project?

- I'm using the convention of mirroring the `src` directory structure within the `__tests__` directory, i.e. the test for `src/defaults/batch.js` is at `__tests__/defaults/batch.test.js`.  The idea behind this being that it makes it easy to know where to look to find a corresponding test for something, but I'm open to any suggestions on the ideal directory structure.

- I've put the `jest-setup.js` file at the root of the project.  Although it's not technically a config file, I tend to think of it like a config file because it is setting up some globals for the Jest environment, so I typically put it at the root of the project, however I think inside of the `__tests__` directory would also be a fine place for it.

Other than that, any feedback on the tests themselves would be great. They're all pretty straight forward unit tests at this point, but if we get these ones to a point where you feel like we're ready to move forward, then I'll finish off the rest of the unit tests and work towards getting some larger integration tests in.
